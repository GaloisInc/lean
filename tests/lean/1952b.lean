structure presheaf_of_types (α : Type*) :=
(F : Π U : set α,  Type*)
(res : ∀ (U V : set α) ,
  (F U) → (F V))
(Hidem : ∀ U : set α, res U U = (res U U) ∘ (res U U))

-- set_option trace.type_context.is_def_eq true
-- set_option trace.type_context.is_def_eq_detail true

definition presheaf_of_types_pushforward
  {β : Type*}
  : presheaf_of_types β :=
  { Hidem := λ U, rfl,
}
