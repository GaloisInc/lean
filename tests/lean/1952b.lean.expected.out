1952b.lean:13:2: error: invalid structure value { ... }, field 'F' was not provided
1952b.lean:13:2: error: invalid structure value { ... }, field 'res' was not provided
1952b.lean:13:13: error: type mismatch at field 'Hidem'
  λ (U : set β), rfl
has type
  ∀ (U : set β), ?m_2[U] = ?m_2[U]
but is expected to have type
  ∀ (U : set β), ⁇ U U = ⁇ U U ∘ ⁇ U U
